front/pages/api/stripe/webhook.ts

```diff
@@ -277,10 +288,60 @@ async function handler(
             { event },
             "[Stripe Webhook] Received customer.subscription.updated event."
           );
+          const previousAttributes = event.data.previous_attributes;
+          if (!previousAttributes) break; // should not happen by definition of the subscription.updated event
+
+          if (
+            // The subscription is canceled (but not yet ended) or reactivated
+            stripeSubscription.status === "active" &&
+            "cancel_at_period_end" in previousAttributes
+          ) {
+            // first update subscription endDate in our database
+            const result = await Subscription.update(
```

Comment:
Instead of directly updating the subscription from the `stripeSubscriptionId`, I would first fetch the subscription object, raise if not found, then update it and then tiny refactor the `getAdminEmailsForSubscription` to pass directly the workspaceId. Then the method could be renamed to something like "getAdminEmailsForWorkspace`. I feel like it would be more consistant with the rest of the codebase. WDYT?

Reply:
Two IOs. But yeah it's better, good point for consistency also. Is it worth the time to refactor? Started to overthink it :thinking: => which made it moot so just did it :+1: